packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -484,17 +484,21 @@ const Tooltip = React.forwardRef(function Tooltip(props, ref) {
     }
   }
 
+  const { popperOptions, ...PopperPropsRest } = PopperProps || {};
+
   // Avoid the creation of a new Popper.js instance at each render.
-  const popperOptions = React.useMemo(
+  const popperOptionsWithArrow = React.useMemo(
     () => ({
+      ...popperOptions,
       modifiers: {
         arrow: {
           enabled: Boolean(arrowRef),
           element: arrowRef,
         },
+        ...(popperOptions && popperOptions.modifiers),
```

Comment:
I wouldn't if we shouldn't be using a deep merge logic (`deepmerge.js`), per the arrow option people might want to override: https://popper.js.org/docs/v2/modifiers/arrow/#padding (but this is v2).

Reply:
Made the switch to using deepmerge.

Reply:
I reverted the commit. deepmerge does not work with DOM elements. arrowRef was causing an infinite recursive loop. I did however add manual merging of default and custom arrow to allow for what you are asking. I do not have the time to look into refactoring deepmerge to accommodate this broken usecase. How about I make an issue to support DOM elements in deepmerge and to switch over to using it at that time?

Reply:
You didn't use the right deepmerge module. Our version should already support DOM elements.

Reply:
Can you clarify which deepmerge module we should be using (I worked with @valgrindMaster on this PR)?
We tried `import deepmerge from `deepmerge`

Reply:
https://github.com/mui-org/material-ui/blob/master/packages/material-ui-utils/src/deepmerge.js